packages/react/src/datepicker/Datepicker.tsx

```diff
@@ -1,18 +1,179 @@
+  const defaultValue = props.defaultSelected ? formatDateToLocalISO(props.defaultSelected) : undefined;
+  const isClient = typeof window === "object";
+  const isMobile = isClient ? window.innerWidth < 480 : false;
+
+  if (isMobile && useNative)
+    return <KVInput type="date" defaultValue={defaultValue} {...KVInputProps} {...commonProps} />;
+
+  return <CustomDatepicker {...props} ref={ref} {...commonProps} />;
+});
+
+const CustomDatepicker = forwardRef<DatepickerPropsWithoutStandard, "input">(
+  (
+    {
+      defaultSelected,
+      defaultMonth,
+      fromDate,
+      toDate,
+      showDropdownMonthYear = true,
+      disableNavigation,
+      showOutsideDays,
+      showWeekNumber,
+      disabledDays,
+      ...KVInputProps
+    },
+    ref,
+  ) => {
+    const [isPickerVisible, setPickerVisible] = useBoolean(false);
+    const { inputProps, dayPickerProps } = useInput({
+      defaultSelected,
+      format: "yyyy-MM-dd",
+      locale: nb,
+      fromDate,
+      toDate,
+    });
 
-export const Datepicker = forwardRef<ChakraInputProps, "input">(
-  ({ size, variant, type = "date", isRequired, isInvalid, isDisabled, ...props }, ref) => {
     return (
-      <ChakraInput
-        {...props}
-        ref={ref}
-        size={size}
-        type={type}
-        variant={variant}
-        isRequired={isRequired}
-        isInvalid={isInvalid}
-        isDisabled={isDisabled}
-      />
+      <Popover
+        placement="bottom-start"
+        isOpen={isPickerVisible}
+        onOpen={setPickerVisible.on}
+        onClose={setPickerVisible.off}
+      >
+        <InputGroup>
+          <PopoverAnchor>
+            <KVInput type="date" ref={ref} className="custom-datepicker" {...KVInputProps} {...inputProps} />
+          </PopoverAnchor>
+
+          <InputRightElement cursor="pointer" onClick={setPickerVisible.toggle} height="100%">
+            <PopoverTrigger>
+              <Icon icon="calendar_today" size={KVInputProps.size === "xs" || KVInputProps.size === "sm" ? 20 : 24} />
+            </PopoverTrigger>
+          </InputRightElement>
+        </InputGroup>
+        <PopoverContent width="auto">
+          <DayPicker
+            captionLayout={showDropdownMonthYear ? "dropdown-buttons" : undefined}
+            disableNavigation={disableNavigation}
+            defaultMonth={defaultMonth}
+            showOutsideDays={showOutsideDays}
+            showWeekNumber={showWeekNumber}
+            disabled={disabledDays}
+            {...dayPickerProps}
+          />
+        </PopoverContent>
+      </Popover>
     );
-  }
+  },
 );
+
+function extractKVProps(props: DatepickerProps): KVInputProps {
+  const {
+    defaultSelected,
+    defaultMonth,
+    fromDate,
+    toDate,
+    showDropdownMonthYear,
+    disableNavigation,
+    showOutsideDays,
+    showWeekNumber,
+    disabledDays,
+    ...chakraProps
+  } = props;
+
+  return chakraProps;
+}
+
+function formatDateToLocalISO(date: Date): string {
```

Comment:
Jeg er skeptisk til å skrive en slik funksjon selv. Tipper javascript har en ferdig funksjon som gjør det du trenger. Og når jeg prøver å velge dato på en av propsene i storybook så feiler storyen. Her har jeg forsøkt å sette defaultSelected:
![bilde](https://github.com/kartverket/kvib/assets/32133041/54578c43-be35-4c4e-bd8f-ee5d56ef2a52)

Reply:
Nå har jeg lagt til feilhåndtering😎 vi må gjøre det på denne måten fordi vi ikke nødvendigvis vet locale og det må vi ha for de innebygde funksjonene. Men nå skal det funke!